src/prefect/agent/agent.py

```diff
@@ -39,6 +41,7 @@ def exit_handler(agent: "Agent") -> Generator:
 
     def _exit_handler(*args: Any, **kwargs: Any) -> None:
         agent.logger.info("Keyboard Interrupt received: Agent is shutting down.")
+        AGENT_WAKE_EVENT.set()
         exit_event.set()
```

Comment:
You probably want to set the `exit_event` first, otherwise there might be a race condition where the agent wakes up, sees it shouldn't exit before the `exit_event` is set. With some logic changes we could get by with only one event, but keeping two is fine for now.

Reply:
You are right. Updated as suggested.